cmake/modules/contrib/TensorRT.cmake

```diff
@@ -28,7 +28,9 @@ if(USE_TENSORRT_CODEGEN)
     file(GLOB RUNTIME_TENSORRT_SRCS src/runtime/contrib/tensorrt/tensorrt_runtime.cc)
     set_source_files_properties(${RUNTIME_TENSORRT_SRCS} PROPERTIES COMPILE_FLAGS "-Wno-deprecated-declarations")
     list(APPEND COMPILER_SRCS ${COMPILER_TENSORRT_SRCS})
-    list(APPEND COMPILER_SRCS ${RUNTIME_TENSORRT_SRCS})
+    if(NOT USE_TENSORRT_RUNTIME)
+        list(APPEND COMPILER_SRCS ${RUNTIME_TENSORRT_SRCS})
```

Comment:
I'm a little confused here. If these are for the runtime, why are they being included in the compiler sources?

Reply:
During codegen, a runtime module needs to be created, so we still need the source file where the custom runtime module is defined so that it can 1) be created and 2) SaveToBinary can be called on it.

https://github.com/apache/tvm/blob/main/src/relay/backend/contrib/tensorrt/codegen.cc#L208
https://github.com/apache/tvm/blob/main/src/relay/backend/contrib/arm_compute_lib/codegen.cc#L370

We could add the files to runtime instead, I don't think it would make a difference.

Reply:
What if we just don't use GLOB here?

Reply:
It needs to be possible to compile with `USE_TENSORRT_CODEGEN` enabled only (and not runtime).

Reply:
Not sure what difference using glob or not would make since only `tensorrt_runtime.cc` is added

Reply:
In the `if(USE_TENSORRT_RUNTIME)` we could not use GLOB and explicitly list the files and not include tensors_runtime.cc. Then we wouldn't need to do the conditional.

But, if USE_TERSORRT_CODEGEN can be disabled while USE_TENSORRT_RUNTIME is enabled, then the current approach is fine.

Reply:
We need to include tensorrt_runtime.cc though for `USE_TENSORRT_CODEGEN`